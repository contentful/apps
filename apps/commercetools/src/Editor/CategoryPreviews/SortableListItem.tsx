import React from "react";
import { SortableElement, SortableHandle } from "react-sortable-hoc";
import { css } from "emotion";
import {
  Card,
  CardDragHandle as FormaCardDragHandle,
  Heading,
  Icon,
  IconButton,
  Subheading,
  Tag,
  Typography
} from "@contentful/forma-36-react-components";
import tokens from "@contentful/forma-36-tokens";
import { Category } from "../../interfaces";

export interface Props {
  category: Category;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const styles = {
  card: css({
    display: "flex",
    padding: 0,
    position: "relative",
    ":not(:first-of-type)": css({
      marginTop: tokens.spacingXs
    })
  }),
  dragHandle: css({
    height: "auto"
  }),
  actions: css({
    position: "absolute",
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: "inline-block",
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`
      }),
      "&:hover": {
        svg: css({
          fill: tokens.colorContrastDark
        })
      }
    })
  }),
  description: css({
    padding: tokens.spacingM,
    flex: "1 0 auto",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center"
  }),
  name: (name?: string) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: tokens.spacing2Xs,
      ...(name && { textTransform: "capitalize" })
    }),
  slug: css({
    color: tokens.colorElementDarkest,
    fontSize: tokens.fontSizeS,
    marginBottom: 0
  })
};

const CardDragHandle = SortableHandle(() => (
  <FormaCardDragHandle className={styles.dragHandle}>
    Reorder category
  </FormaCardDragHandle>
));

export const SortableListItem = SortableElement<Props>(
  ({ category, disabled, isSortable, onDelete }: Props) => {
    const categoryIsMissing = !category.name;

    return (
      <Card className={styles.card}>
        <>
          {isSortable && <CardDragHandle />}
          <section className={styles.description}>
            <Typography>
              <Heading className={styles.name(category.name)}>
                {categoryIsMissing ? category.id : category.name}
              </Heading>
              {categoryIsMissing ? (
                <Tag tagType="negative">Category missing</Tag>
              ) : (
                <Subheading className={styles.slug}>{category.slug}</Subheading>
              )}
            </Typography>
          </section>
        </>
        {!disabled && (
          <div className={styles.actions}>
            {category.externalLink && (
              <a
                target="_blank"
                rel="noopener noreferrer"
                href={category.externalLink}
              >
                <Icon icon="ExternalLink" color="muted" />
              </a>
            )}
            <IconButton
              label="Delete"
              iconProps={{ icon: "Close" }}
              {...{
                buttonType: "muted",
                onClick: onDelete
              }}
            />
          </div>
        )}
      </Card>
    );
  }
);
