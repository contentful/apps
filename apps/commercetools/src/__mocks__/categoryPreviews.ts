import { Category } from './../interfaces';

const categoryPreviews: Category[] = [
  {
    id: '28a7442d',
    name: "Mens' clothes",
    slug: 'men-regular',
    externalLink: 'https://mc.commercetools.com/example-project/categories/28a7442d/general'
  },
  {
    id: '38a7442d',
    name: "Womens' clothes",
    slug: 'women-regular',
    externalLink: 'https://mc.commercetools.com/example-project/categories/38a7442d/general'
  },
  {
    id: '48a7442d',
    name: "Kids' clothes",
    slug: 'kids-regular',
    externalLink: 'https://mc.commercetools.com/example-project/categories/48a7442d/general'
  }
];

export default categoryPreviews;
