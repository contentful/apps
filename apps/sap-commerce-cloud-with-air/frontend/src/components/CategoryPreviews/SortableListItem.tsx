import { ReactElement } from 'react';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import { css } from 'emotion';
import { Badge, Card, Heading, IconButton, Subheading } from '@contentful/f36-components';
import { CloseIcon, ExternalLinkIcon } from '@contentful/f36-icons';
import tokens from '@contentful/forma-36-tokens';
import { Category } from '../../interfaces';

export interface Props {
  category: Category;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const styles = {
  card: css({
    display: 'flex',
    padding: 0,
    position: 'relative',
    ':not(:first-of-type)': css({
      marginTop: tokens.spacingXs,
    }),
  }),
  cardInner: css({
    display: 'flex',
  }),
  dragHandle: css({
    height: 'auto',
  }),
  actions: css({
    position: 'absolute',
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: 'inline-block',
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`,
      }),
      '&:hover': {
        svg: css({
          fill: tokens.colorBlack,
        }),
      },
    }),
  }),
  description: css({
    padding: tokens.spacingM,
    flex: '1 0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  }),
  heading: (category: Category) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: category.isMissing || !category.name ? 0 : tokens.spacing2Xs,
      ...(category.name && { textTransform: 'capitalize' }),
    }),
  subheading: css({
    color: tokens.gray500,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
  slug: css({
    color: tokens.colorBlack,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
};

const CardDragHandle = SortableHandle(({ drag }: { drag: ReactElement }) => <>{drag}</>);

function getCategoryIdentifier(category: Category) {
  return category.slug.length ? category.slug : category.id;
}

export const SortableListItem = SortableElement<Props>(
  ({ category, disabled, isSortable, onDelete }: Props) => {
    return (
      <Card
        className={styles.card}
        withDragHandle
        dragHandleRender={isSortable ? ({ drag }) => <CardDragHandle drag={drag} /> : undefined}>
        <div className={styles.cardInner}>
          <section className={styles.description}>
            <>
              <Heading className={styles.heading(category)}>
                {category.isMissing || !category.name
                  ? getCategoryIdentifier(category)
                  : category.name}
              </Heading>
              {category.isMissing && <Badge variant="negative">Category missing</Badge>}
              {!category.isMissing && category.name && (
                <Subheading className={styles.subheading}>
                  {getCategoryIdentifier(category)}
                </Subheading>
              )}
            </>
          </section>
        </div>
        {!disabled && (
          <div className={styles.actions}>
            {category.externalLink && (
              <a target="_blank" rel="noopener noreferrer" href={category.externalLink}>
                <ExternalLinkIcon color="muted" />
              </a>
            )}
            <IconButton
              aria-label="Delete"
              icon={<CloseIcon />}
              variant="transparent"
              onClick={onDelete}
            />
          </div>
        )}
      </Card>
    );
  },
);
