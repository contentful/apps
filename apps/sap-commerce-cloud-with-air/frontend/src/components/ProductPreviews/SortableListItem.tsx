import { useState, ReactElement } from 'react';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import { css } from 'emotion';
import {
  Badge,
  Card,
  Heading,
  IconButton,
  SkeletonContainer,
  SkeletonImage,
  Subheading,
} from '@contentful/f36-components';
import { AssetIcon, CloseIcon, ErrorCircleIcon, ExternalLinkIcon } from '@contentful/f36-icons';
import tokens from '@contentful/forma-36-tokens';
import { Product } from '../../interfaces';

export interface Props {
  product: Product;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const IMAGE_SIZE = 48;

const styles = {
  card: css({
    padding: 0,
    position: 'relative',
    ':not(:first-of-type)': css({
      marginTop: tokens.spacingXs,
    }),
  }),
  cardInner: css({
    display: 'flex',
  }),
  imageWrapper: (imageHasLoaded: boolean) =>
    css({
      width: imageHasLoaded ? `${IMAGE_SIZE}px` : 0,
      height: imageHasLoaded ? `${IMAGE_SIZE}px` : 0,
      overflow: 'hidden',
      margin: imageHasLoaded ? tokens.spacingM : 0,
      position: 'relative',
      '> img': css({
        display: 'block',
        height: `${IMAGE_SIZE}px`,
        minWidth: 'auto',
        userSelect: 'none',
        position: 'absolute',
        left: '50%',
        top: '50%',
        transform: 'translate(-50%, -50%)',
      }),
    }),
  dragHandle: css({
    height: 'auto',
    borderBottomLeftRadius: tokens.borderRadiusMedium,
    borderTopLeftRadius: tokens.borderRadiusMedium,
    cursor: 'grab',
  }),
  actions: css({
    position: 'absolute',
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: 'inline-block',
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`,
      }),
      '&:hover': {
        svg: css({
          fill: tokens.colorBlack,
        }),
      },
    }),
  }),
  description: css({
    flex: '1 0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  }),
  heading: (product: Product) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: product.isMissing || !product.name ? 0 : tokens.spacing2Xs,
      ...(product.name && { textTransform: 'capitalize' }),
    }),
  subheading: css({
    color: tokens.gray500,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
  skeletonImage: css({
    width: `${IMAGE_SIZE}px`,
    height: `${IMAGE_SIZE}px`,
    padding: tokens.spacingM,
  }),
  errorImage: css({
    backgroundColor: tokens.gray100,
    borderRadius: '3px',
    margin: tokens.spacingM,
    width: `${IMAGE_SIZE}px`,
    height: `${IMAGE_SIZE}px`,
    position: 'relative',
    svg: css({
      fill: tokens.gray600,
      width: '100%',
      height: '50%',
      position: 'absolute',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
    }),
  }),
};

const CardDragHandle = SortableHandle(({ drag }: { drag: ReactElement }) => <>{drag}</>);

export const SortableListItem = SortableElement<Props>(
  ({ product, disabled, isSortable, onDelete }: Props) => {
    const [imageHasLoaded, setImageLoaded] = useState(false);
    const [imageHasErrored, setImageHasErrored] = useState(false);

    return (
      <Card
        className={styles.card}
        withDragHandle
        dragHandleRender={isSortable ? ({ drag }) => <CardDragHandle drag={drag} /> : undefined}>
        <div className={styles.cardInner}>
          {!imageHasLoaded && !imageHasErrored && (
            <SkeletonContainer className={styles.skeletonImage}>
              <SkeletonImage width={IMAGE_SIZE} height={IMAGE_SIZE} />
            </SkeletonContainer>
          )}
          {imageHasErrored && (
            <div className={styles.errorImage}>
              {product.isMissing ? (
                <ErrorCircleIcon testId="error-circle-icon" />
              ) : (
                <AssetIcon testId="asset-icon" />
              )}
            </div>
          )}
          {!imageHasErrored && (
            <div className={styles.imageWrapper(imageHasLoaded)}>
              <img
                style={{ display: imageHasLoaded ? 'block' : 'none' }}
                onLoad={() => setImageLoaded(true)}
                onError={() => setImageHasErrored(true)}
                src={product.image}
                alt={product.name}
                data-test-id="image"
              />
            </div>
          )}
          <section className={styles.description}>
            <Heading className={styles.heading(product)}>
              {product.isMissing || !product.name ? product.sku : product.name}
            </Heading>
            {product.isMissing && <Badge variant="negative">Product missing</Badge>}
            {!product.isMissing && product.name && (
              <Subheading className={styles.subheading}>{product.sku}</Subheading>
            )}
          </section>
        </div>
        {!disabled && (
          <div className={styles.actions}>
            {product.externalLink && (
              <a target="_blank" rel="noopener noreferrer" href={product.externalLink}>
                <ExternalLinkIcon color="muted" />
              </a>
            )}
            <IconButton
              aria-label="Delete"
              icon={<CloseIcon />}
              variant="transparent"
              onClick={onDelete}
            />
          </div>
        )}
      </Card>
    );
  },
);
