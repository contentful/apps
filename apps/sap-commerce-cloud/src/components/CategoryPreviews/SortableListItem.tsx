import React from 'react';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import { css } from 'emotion';
import {
  Card,
  CardDragHandle as FormaCardDragHandle,
  Heading,
  Icon,
  IconButton,
  Subheading,
  Tag,
  Typography,
} from '@contentful/forma-36-react-components';
import tokens from '@contentful/forma-36-tokens';
import { Category } from '../../interfaces';

export interface Props {
  category: Category;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const styles = {
  card: css({
    display: 'flex',
    padding: 0,
    position: 'relative',
    ':not(:first-of-type)': css({
      marginTop: tokens.spacingXs,
    }),
  }),
  dragHandle: css({
    height: 'auto',
  }),
  actions: css({
    position: 'absolute',
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: 'inline-block',
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`,
      }),
      '&:hover': {
        svg: css({
          fill: tokens.colorContrastDark,
        }),
      },
    }),
  }),
  description: css({
    padding: tokens.spacingM,
    flex: '1 0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  }),
  heading: (category: Category) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: category.isMissing || !category.name ? 0 : tokens.spacing2Xs,
      ...(category.name && { textTransform: 'capitalize' }),
    }),
  subheading: css({
    color: tokens.colorElementDarkest,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
  slug: css({
    color: tokens.colorElementDarkest,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
};

const CardDragHandle = SortableHandle(() => (
  <FormaCardDragHandle className={styles.dragHandle}>Reorder category</FormaCardDragHandle>
));

function getCategoryIdentifier(category: Category) {
  return category.slug.length ? category.slug : category.id;
}

export const SortableListItem = SortableElement<Props>(
  ({ category, disabled, isSortable, onDelete }: Props) => {
    return (
      <Card className={styles.card}>
        <>
          {isSortable && <CardDragHandle />}
          <section className={styles.description}>
            <Typography>
              <Heading className={styles.heading(category)}>
                {category.isMissing || !category.name
                  ? getCategoryIdentifier(category)
                  : category.name}
              </Heading>
              {category.isMissing && <Tag tagType="negative">Category missing</Tag>}
              {!category.isMissing && category.name && (
                <Subheading className={styles.subheading}>
                  {getCategoryIdentifier(category)}
                </Subheading>
              )}
            </Typography>
          </section>
        </>
        {!disabled && (
          <div className={styles.actions}>
            {category.externalLink && (
              <a target="_blank" rel="noopener noreferrer" href={category.externalLink}>
                <Icon icon="ExternalLink" color="muted" />
              </a>
            )}
            <IconButton
              label="Delete"
              iconProps={{ icon: 'Close' }}
              buttonType="muted"
              onClick={onDelete}
            />
          </div>
        )}
      </Card>
    );
  }
);
