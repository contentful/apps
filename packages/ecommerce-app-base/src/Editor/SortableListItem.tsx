import {
  Badge,
  Card,
  Heading,
  IconButton,
  SkeletonContainer,
  SkeletonImage,
  Subheading,
} from '@contentful/f36-components';
import { AssetIcon, CloseIcon, ErrorCircleIcon, ExternalLinkIcon } from '@contentful/f36-icons';
import tokens from '@contentful/f36-tokens';
import { css } from 'emotion';
import React, { ReactElement, useState } from 'react';
import { SortableElement, SortableHandle } from 'react-sortable-hoc';
import { Product } from '../interfaces';

export interface Props {
  product: Product;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
  skuType?: string;
}

const IMAGE_SIZE = 48;

const styles = {
  card: css({
    padding: 0,
    position: 'relative',
    ':not(:first-of-type)': css({
      marginTop: tokens.spacingXs,
    }),
  }),
  cardInner: css({
    display: 'flex',
  }),
  imageWrapper: (imageHasLoaded: boolean) =>
    css({
      width: imageHasLoaded ? `${IMAGE_SIZE}px` : 0,
      height: imageHasLoaded ? `${IMAGE_SIZE}px` : 0,
      overflow: 'hidden',
      margin: imageHasLoaded ? tokens.spacingM : 0,
      position: 'relative',
      '> img': css({
        display: 'block',
        height: `${IMAGE_SIZE}px`,
        minWidth: 'auto',
        userSelect: 'none',
        position: 'absolute',
        left: '50%',
        top: '50%',
        transform: 'translate(-50%, -50%)',
      }),
    }),
  dragHandle: css({
    height: 'auto',
    borderBottomLeftRadius: tokens.borderRadiusMedium,
    borderTopLeftRadius: tokens.borderRadiusMedium,
    cursor: 'grab',
  }),
  actions: css({
    position: 'absolute',
    top: tokens.spacingXs,
    right: tokens.spacingXs,
    a: css({
      display: 'inline-block',
      marginRight: tokens.spacingXs,
      svg: css({
        transition: `fill ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault}`,
      }),
      '&:hover': {
        svg: css({
          fill: tokens.colorBlack,
        }),
      },
    }),
  }),
  description: css({
    flex: '1 0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
  }),
  name: (name?: string) =>
    css({
      fontSize: tokens.fontSizeL,
      marginBottom: tokens.spacing2Xs,
      ...(name && { textTransform: 'capitalize' }),
    }),
  sku: css({
    color: tokens.gray500,
    fontSize: tokens.fontSizeS,
    marginBottom: 0,
  }),
  skeletonImage: css({
    width: `${IMAGE_SIZE}px`,
    height: `${IMAGE_SIZE}px`,
    padding: tokens.spacingM,
  }),
  errorImage: css({
    backgroundColor: tokens.gray100,
    borderRadius: '3px',
    margin: tokens.spacingM,
    width: `${IMAGE_SIZE}px`,
    height: `${IMAGE_SIZE}px`,
    position: 'relative',
    svg: css({
      fill: tokens.gray600,
      width: '100%',
      height: '50%',
      position: 'absolute',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
    }),
  }),
};

const CardDragHandle = SortableHandle(({ drag }: { drag: ReactElement }) => <>{drag}</>);

export const SortableListItem = SortableElement<Props>(
  ({ product, disabled, isSortable, onDelete, skuType }: Props) => {
    const [imageHasLoaded, setImageLoaded] = useState(false);
    const [imageHasErrored, setImageHasErrored] = useState(false);
    const productIsMissing = !product.name;

    return (
      <Card
        data-testid="sortable-list-item"
        className={styles.card}
        // @ts-ignore
        withDragHandle
        dragHandleRender={isSortable ? ({ drag }) => <CardDragHandle drag={drag} /> : undefined}
      >
        <div className={styles.cardInner}>
          {!imageHasLoaded && !imageHasErrored && product.image && (
            <SkeletonContainer className={styles.skeletonImage}>
              <SkeletonImage width={IMAGE_SIZE} height={IMAGE_SIZE} />
            </SkeletonContainer>
          )}
          {!product.image || imageHasErrored ? (
            <div className={styles.errorImage}>
              {productIsMissing ? (
                <ErrorCircleIcon testId="error-circle-icon" />
              ) : (
                <AssetIcon testId="asset-icon" />
              )}
            </div>
          ) : (
            <div className={styles.imageWrapper(imageHasLoaded)}>
              <img
                style={{ display: imageHasLoaded ? 'block' : 'none' }}
                onLoad={() => setImageLoaded(true)}
                onError={() => setImageHasErrored(true)}
                src={product.image}
                alt={product.name}
                data-test-id="image"
              />
            </div>
          )}
          <section className={styles.description}>
            <Heading className={styles.name(product.name)}>
              {productIsMissing ? product.sku : product.name}
            </Heading>
            {productIsMissing ? (
              <Badge variant="negative">{skuType ?? 'Product'} missing</Badge>
            ) : (
              <Subheading className={styles.sku}>{product.displaySKU ?? product.sku}</Subheading>
            )}
          </section>
        </div>
        {!disabled && (
          <div className={styles.actions}>
            {product.externalLink && (
              <a target="_blank" rel="noopener noreferrer" href={product.externalLink}>
                <ExternalLinkIcon color="muted" />
              </a>
            )}
            <IconButton
              aria-label="Delete"
              icon={<CloseIcon />}
              variant="transparent"
              onClick={onDelete}
            />
          </div>
        )}
      </Card>
    );
  }
);
