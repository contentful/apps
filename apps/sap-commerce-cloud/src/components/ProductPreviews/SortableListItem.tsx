import { FC, useState } from 'react';
import { useSortable } from '@dnd-kit/sortable';
import { CSS } from '@dnd-kit/utilities';
import { Product } from '../../interfaces';
import {
  AssetIcon,
  Badge,
  Card,
  Flex,
  Heading,
  IconButton,
  SkeletonContainer,
  SkeletonImage,
  Subheading,
} from '@contentful/f36-components';
import tokens from '@contentful/f36-tokens';
import { styles } from './SortableListItem.styles';
import { CloseIcon, ErrorCircleIcon, ExternalLinkIcon } from '@contentful/f36-icons';

export interface Props {
  product: Product;
  disabled: boolean;
  onDelete: () => void;
  isSortable: boolean;
}

const IMAGE_SIZE = 48;

export const SortableListItem: FC<Props> = ({ product, disabled, onDelete, isSortable }: Props) => {
  const [imageHasLoaded, setImageLoaded] = useState(false);
  const [imageHasErrored, setImageHasErrored] = useState(false);

  const { attributes, listeners, setNodeRef, transform, transition } = useSortable({
    id: product.productUrl,
  });

  const style = isSortable
    ? {
        transform: CSS.Transform.toString(transform),
        transition,
        marginTop: tokens.spacingS,
      }
    : { marginTop: tokens.spacingS };

  return (
    <div
      ref={setNodeRef}
      style={style}
      key={product.id}
      {...attributes}
      {...listeners}
      data-testid={`sortable-item-${product.id}`}>
      <Card withDragHandle={isSortable} className={styles.card} data-testid="sortable-list-item">
        <Flex>
          {!imageHasLoaded && !imageHasErrored && (
            <SkeletonContainer className={styles.skeletonImage}>
              <SkeletonImage width={IMAGE_SIZE} height={IMAGE_SIZE} />
            </SkeletonContainer>
          )}
          {imageHasErrored && (
            <div className={styles.errorImage}>
              {product.isMissing ? (
                <ErrorCircleIcon data-testid="icon" />
              ) : (
                <AssetIcon data-testid="icon" />
              )}
            </div>
          )}
          {!imageHasErrored && (
            <div className={styles.imageWrapper(imageHasLoaded)}>
              <img
                style={{ display: imageHasLoaded ? 'block' : 'none' }}
                onLoad={() => setImageLoaded(true)}
                onError={() => setImageHasErrored(true)}
                src={product.image}
                alt={product.name}
                data-testid="image"
              />
            </div>
          )}
          <section className={styles.description}>
            <Heading className={styles.heading(product)}>
              {product.isMissing || !product.name ? product.sku : product.name}
            </Heading>
            {product.isMissing && <Badge variant="negative">Product missing</Badge>}
            {!product.isMissing && product.name && (
              <Subheading className={styles.subheading}>{product.sku}</Subheading>
            )}
          </section>
          {!disabled && (
            <div className={styles.actions}>
              {product.externalLink && (
                <a target="_blank" rel="noopener noreferrer" href={product.externalLink}>
                  <ExternalLinkIcon variant="muted" />
                </a>
              )}
              <IconButton
                icon={<CloseIcon variant="muted" />}
                aria-label="Delete"
                onClick={onDelete}
              />
            </div>
          )}
        </Flex>
      </Card>
    </div>
  );
};
