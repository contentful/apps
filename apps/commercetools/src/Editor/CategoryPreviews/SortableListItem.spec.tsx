import React from 'react';
import identity from 'lodash/identity';
import { configure, render, cleanup } from '@testing-library/react';
import { Props, SortableListItem } from './SortableListItem';
import categoryPreviews from '../../__mocks__/categoryPreviews';

configure({
  testIdAttribute: 'data-test-id'
});

const defaultProps: Props = {
  category: categoryPreviews[0],
  disabled: false,
  onDelete: jest.fn(),
  isSortable: false
};

const renderComponent = (props: Props) => {
  return render(<SortableListItem index={0} {...props} />);
};

jest.mock('react-sortable-hoc', () => ({
  SortableContainer: identity,
  SortableElement: identity,
  SortableHandle: identity
}));

describe('SortableListItem', () => {
  afterEach(cleanup);

  it('should render successfully', async () => {
    const component = renderComponent(defaultProps);
    expect(component.container).toMatchSnapshot();
  });

  it('should render successfully the error variation for missing slug', () => {
    const component = renderComponent({
      ...defaultProps,
      category: { ...categoryPreviews[0], name: '' }
    });
    expect(component.container).toMatchSnapshot();
  });
});
