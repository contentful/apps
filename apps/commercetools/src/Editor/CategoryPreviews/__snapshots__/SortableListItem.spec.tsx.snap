// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortableListItem should render successfully 1`] = `
<div>
  <div
    class="Card__Card___xXJsT css-pivbrd Card__Card--padding-default___b6Brl"
    data-test-id="cf-ui-card"
  >
    <section
      class="css-ijlwe4"
    >
      <div
        class="Typography__Typography___1ZUfE"
        data-test-id="cf-ui-text-container"
      >
        <h1
          class="Heading__Heading___83D3K css-1xd3m0r f36-margin-bottom--m"
          data-test-id="cf-ui-heading"
        >
          Mens' clothes
        </h1>
        <h2
          class="Subheading__Subheading___2mA9j css-1b5bj2l f36-margin-bottom--m"
          data-test-id="cf-ui-subheading"
        >
          men-regular
        </h2>
      </div>
    </section>
    <div
      class="css-zdzpuy"
    >
      <a
        href="https://mc.commercetools.com/example-project/categories/28a7442d/general"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </a>
      <button
        class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
        data-test-id="cf-ui-icon-button"
        type="button"
      >
        <span
          class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
          tabindex="-1"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK undefined IconButton__IconButton__icon___3yZQN"
            data-test-id="cf-ui-icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          <span
            class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
          >
            Delete
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`SortableListItem should render successfully the error variation for missing slug 1`] = `
<div>
  <div
    class="Card__Card___xXJsT css-pivbrd Card__Card--padding-default___b6Brl"
    data-test-id="cf-ui-card"
  >
    <section
      class="css-ijlwe4"
    >
      <div
        class="Typography__Typography___1ZUfE"
        data-test-id="cf-ui-text-container"
      >
        <h1
          class="Heading__Heading___83D3K css-k5fxdf f36-margin-bottom--m"
          data-test-id="cf-ui-heading"
        >
          men-regular
        </h1>
        
      </div>
    </section>
    <div
      class="css-zdzpuy"
    >
      <a
        href="https://mc.commercetools.com/example-project/categories/28a7442d/general"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </a>
      <button
        class="IconButton__IconButton___1_XeU a11y__focus-outline--default___2hwb1 IconButton__IconButton--muted___22_IZ"
        data-test-id="cf-ui-icon-button"
        type="button"
      >
        <span
          class="TabFocusTrap__TabFocusTrap___39Vty IconButton__IconButton__inner___3fnmT"
          tabindex="-1"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK undefined IconButton__IconButton__icon___3yZQN"
            data-test-id="cf-ui-icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          <span
            class="IconButton__IconButton__label___1kp5y helpers__sr-only___3Kv3z"
          >
            Delete
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
